### ISSUE TITLE
Loading PHPMailer 6.0

### ISSUE THREAD

**Issue by ridera** [PHPMailer/PHPMailer#959]
Having problem with upgrading phpMailer 5.xx to 6.0. Used several versions of phpmailer for years, no problems like this. I prefer to not use composer.

From upgrade guide "Alternatively, if you're not using composer, copy the contents of the PHPMailer folder into one of the include_path directories specified in your PHP configuration and load each one manually." I prefer to not modify the include_path file on a shared host.

I used the PHPMailerAutoload.php for my 5.xx versions

I used this from the 6.0 package: "Upgrading from PHPMailer 5.2 to 6.0 If you're not using composer, you can still load the classes manually, depending on what you're using:" ```

use PHPMailer\PHPMailer\PHPMailer;
use PHPMailer\PHPMailer\Exception;

require 'src/PHPMailer.php';
require 'src/SMTP.php';
// No errors reported on these

$mail = new PHPMailer(true);
Fatal error: Uncaught Error: Class 'PHPMailer' not found in ///path to my code

Tried $mail = new \PHPMailer(true);
Tried lower case file names for PHPMailer.php and SMTP.php
Tried class.PHPMailer.php and class.SMTP.php; and with lowercase filenames

**Comment by Synchro** [comment]
This is just namespacing. Look at the examples for reference - you need to either `use` the namespaced class or reference it absolutely, for example:

    use PHPMailer\PHPMailer\PHPMailer;
    $mail = new PHPMailer(true);

or

    $mail = new use PHPMailer\PHPMailer\PHPMailer(true);

**Comment by Synchro** [comment]
Just to confirm, I altered the beginning of the gmail example to look like this, and it worked fine:

```php
use PHPMailer\PHPMailer\PHPMailer;
use PHPMailer\PHPMailer\SMTP;
use PHPMailer\PHPMailer\Exception;

require '../src/PHPMailer.php';
require '../src/SMTP.php';
require '../src/Exception.php';

//Create a new PHPMailer instance
$mail = new PHPMailer;
```

**Comment by ridera** [comment]
No work   I copied your lines to make certain of no typo errors, 
These are in my config file, which is called before the functions file
use PHPMailer\PHPMailer\PHPMailer;
use PHPMailer\PHPMailer\SMTP;
use PHPMailer\PHPMailer\Exception;

  require MAILER_DIR .'/src/PHPMailer.php';
  require MAILER_DIR .'/src/SMTP.php';
  require MAILER_DIR .'/src/Exception.php';
No errors on these requires

This is in my functions file
$mail = new PHPMailer;  I even took out the (test)
same error 
<b>Fatal error</b>:  Uncaught Error: Class 'PHPMailer' not found in .....
using PHP Ver: 7.1.0;

**Comment by ridera** [comment]
I replied this via GitHub

No work I copied your lines to make certain of no typo errors,

These are in my config file, which is called before the functions file
use PHPMailer\PHPMailer\PHPMailer;
use PHPMailer\PHPMailer\SMTP;
use PHPMailer\PHPMailer\Exception;

require MAILER_DIR .'/src/PHPMailer.php';
require MAILER_DIR .'/src/SMTP.php';
require MAILER_DIR .'/src/Exception.php';
No errors on these requires

This is in my functions file
$mail = new PHPMailer; I even took out the (test)

same error
*Fatal error*: Uncaught Error: Class 'PHPMailer' not found in .....
using PHP Ver: 7.1.0;

**Comment by Synchro** [comment]
Ah, well that's a PHP issue. `Use` statements only apply in the file in which they are defined. As [the docs say](http://php.net/manual/en/language.namespaces.importing.php):

> Importing rules are per file basis, meaning included files will NOT inherit the parent file's importing rules.

If you move the `use` statements to your functions file, it will work.

**Comment by ridera** [comment]
Many thanks. That worked

Just so I'm clear.  Does version 6 **require** the use of namespace?  If so, would be good to make this point explicit in the upgrade [Upgrading from PHPMailer 5.2 to 6.0] and examples files.

**Comment by Synchro** [comment]
Yes - The classes are namespaced, so any references you make to them must respect that, just like anything that uses a namespace. It's is covered pretty thoroughly in [the upgrade guide](https://github.com/PHPMailer/PHPMailer/blob/6.0/UPGRADING.md), which is linked from the readme, though the link doesn't work because it's expecting it to be in master, which it's not, yet. It's already reflected in all the example files in the 6.0 branch, though obviously not in the 5.2/master branch.

**Comment by ridera** [comment]
I saw the "You *will need to* import classes you're using explicitly into your own namespace, or the global namespace - 
all the examples show how to do this."

In my opinion, this is not very explicit.  Suggest rewording to read *"*You *Must *import classes you're using 
explicitly into your own namespace, or the global namespace.

I believe this feature will cause a lot of grief to others as they attempt to upgrade the phpMailer 5.xx installations 
to 6.xx.  Once I got the classes to load the 6.0, my 5.2.21 smtp mode settings ran perfect without any changes.

Seems to me the use of namespace should be an option, not a requirement.

Al....

**Comment by Synchro** [comment]
Unfortunately you can't have it both ways - it's all or nothing. Namespacing is pretty important for libraries - there have been quite a few issues caused by class name clashes with SMTP in particular, it's simply expected of any modern library, and it's probably the single most requested feature.

This issue isn't specific to PHPMailer, it's exactly the same with all namespaced libraries & frameworks, which now covers Symfony, Zend, Laravel, Drupal, and pretty much everything else now. In your case you were doing something that would break any namespaced library, not just PHPMailer. All this kind of messing about with loading classes and defining things in the right place manually is precisely why composer exists; if you're using composer, you can pretty much forget about ever needing to know where a class file is.

Anyway, I can make the wording a little stronger to make it unambiguous.

**Comment by andrealansp** [comment]
Hello, I used the composer for install phpmailer, on my project, but when i call de autoload i receive the error  Uncaught Error: Class 'PHPMailer' not found in C:\wamp64\www\php\tpl\email.php:24 Stack trace: #0 {main} thrown in C:\wamp64\www\php\tpl\email.php on line 24. 

What do i can do ?

Thanks for help!

**Comment by Synchro** [comment]
I can't tell from what you posted. Do what it says in the readme, base your code on the examples provided.

**Comment by sunil-tamang** [comment]
Fatal error: Uncaught Error: Class 'PHPMailer\PHPMailer' not found
 
 used this  $mail = new PHPMailer(true); 

almost tried everythin help......

**Comment by Synchro** [comment]
Read this thread, do what it says in the readme, base your code on the examples provided.

**Comment by ccyarm** [comment]
I've got a problem with "use" in my mailer.php program
Parse error: syntax error, unexpected 'use' (T_USE) in D:\test\common\mailer.php on line 25
What am I missing?
Thank you for any help.

**Comment by sumiah4** [comment]
Hi synchro,

I have the same problem.And I am navigating to this page using ajax.I installed it manually unziped the file and just added to my folder.
use PHPMailer\PHPMailer\PHPMailer;
use PHPMailer\PHPMailer\Exception;

require ('PHPMailer/src/PHPMailer.php');
require ('PHPMailer/src/Exception.php');


$mail = new \PHPMailer(true);
I did all the way i could still the error alerts like
Uncaught Error: Class 'PHPMailer'.Please help .What to do?

**Comment by Synchro** [comment]
Do what it says in this thread. You have exactly the same problem and should use the same solution.

**Comment by sumiah4** [comment]
I tried everypart.
Now it says 'Uncaught PHPMailer\PHPMailer\Exception:' when i changed the script to
use PHPMailer\PHPMailer\Exception;
require ('include/PHPMailer/src/Exception.php');
require ('include/PHPMailer/src/PHPMailer.php');

	$rid=$superCage->post->getEscaped('rid');
	$mail = new PHPMailer\PHPMailer\PHPMailer(true);

Am I doing it right?

**Comment by Synchro** [comment]
Yes, except that you have asked PHPMailer to throw exceptions (by passing `true` to the constructor), but then you are not catching the exceptions (hence the uncaught exception error). Look at the exceptions example for how to handle this.

**Comment by sumiah4** [comment]
When I tried exception error catch i git the error like 'syntax error, unexpected 'use' (T_USE)',I am not sure what to do?

**Comment by Synchro** [comment]
Just like the examples say:

```php
<?php
use PHPMailer\PHPMailer\PHPMailer;
use PHPMailer\PHPMailer\Exception;
require ('include/PHPMailer/src/Exception.php');
require ('include/PHPMailer/src/PHPMailer.php');

$mail = new PHPMailer(true);
try {
    //The rest of your mail sending code goes here
} catch (Exception $e) {
    echo 'Error: ', $e->getMessage();
}
```

If your code fails on the first `use` statement, it means you are using a very old version of PHP, and you should upgrade.

**Comment by sumiah4** [comment]
ok.I checked and found it to be the latest version.that is PHP Version 7.1.29.This is the latest version.

**Comment by Synchro** [comment]
[Current PHP](https://php.net/) is 7.3.6, but 7.1.x is new enough to have support for `use`, so you must be doing something else in your code.

**Comment by sumiah4** [comment]
OK.Thanks for your response.Iwill recheck it

**Comment by HrDv** [comment]
Hi Synchro, Hi everybody. I'm a new member. Mojave OS X. MAMP php7.3.1. Google emails. 

I created a folder named "email" inside htdocs. Inside this folder, I created email.php (directly from your simple example) and I placed manually the folder "PHPMailer-master" and "vendor". I've just created a new google account for my trials : "account2@gmail.com" with his password "pwdAccount2". To send an email-test to my previous email "account1@gmail.com". 

The result : the mail is told sent. Was severely blocked by google, with about 20 lines of error from PHP mailer. Here is my code :

<?php
use PHPMailer\PHPMailer\PHPMailer;
use PHPMailer\PHPMailer\SMTP;
use PHPMailer\PHPMailer\Exception;

require 'PHPMailer-master/src/PHPMailer.php';
require 'PHPMailer-master/src/SMTP.php';
require 'PHPMailer-master/src/Exception.php';

//Create a new PHPMailer instance
$mail = new PHPMailer;

try {
	//Server settings
	$mail->SMTPDebug = 2;                                       // Enable verbose debug output
	$mail->isSMTP();                                            // Set mailer to use SMTP
	$mail->Host       = 'smtp.gmail.com';  						// Specify main and backup SMTP servers
	$mail->SMTPAuth   = true;                                   // Enable SMTP authentication
	$mail->Username   = 'account2@gmail.com';                   // SMTP username
	$mail->Password   = 'pwdAccount2';                          // SMTP password
	$mail->SMTPSecure = 'tls';                                  // Enable TLS encryption, `ssl` also accepted
	$mail->Port       = 587;                                    // TCP port to connect to

	//Recipients
	$mail->setFrom('account2@gmail.com', 'Mailer');
	$mail->addAddress('account1.com', 'Joe User');     // Add a recipient
//	$mail->addAddress('ellen@example.com');               // Name is optional
//	$mail->addReplyTo('info@example.com', 'Information');
//	$mail->addCC('cc@example.com');
//	$mail->addBCC('bcc@example.com');
	
	// Attachments
//	$mail->addAttachment('/var/tmp/file.tar.gz');         // Add attachments
//	$mail->addAttachment('/tmp/image.jpg', 'new.jpg');    // Optional name
	
	// Content
	$mail->isHTML(true);                                  // Set email format to HTML
	$mail->Subject = 'Here is the subject';
	$mail->Body    = 'This is the HTML message body <b>in bold!</b>';
	$mail->AltBody = 'This is the body in plain text for non-HTML mail clients';
	
	$mail->send();
	echo 'Message has been sent';
} catch (Exception $e) {
    echo "Message could not be sent. Mailer Error: {$mail->ErrorInfo}";
}
?>

**Comment by HrDv** [comment]
didn't work for importing code, I try another way
[email.docx](https://github.com/PHPMailer/PHPMailer/files/3422807/email.docx)

**Comment by HrDv** [comment]
Thank you for Help

**Comment by Synchro** [comment]
Welcome to Github. Please don't post comments on old, closed issues, especially when they are not related to your problem - it sends notifications to everyone that has ever been part of the thread.

It sounds like your script is working fine, but you have debug output enabled, so turn it off by setting `SMTPDebug = 0`.

**Comment by HrDv** [comment]
Sorry, 'didnt' see the "closed" mention at the top. I'll try your new setting, thank you

**Comment by HrDv** [comment]
Thank dear Marcus Bointon. I'll take care next about the closed topics.

At the moment setting SMTPDebut to 0 just makes the errors disappearing but
did not cancel the severe google mail blocking nor probably the error from
PHP Mailer (just masked). No email received. Nor sent.

I tried unsuccessfully you gmail example furnished. May be my MAMP-inside
working ? Outside composer. Outside user/local/... So I have to work hard
first and seek deeper about my bugs before posting again.

Hery

Le mar. 23 juil. 2019 à 18:03, Marcus Bointon <notifications@github.com> a
écrit :

**Comment by Synchro** [comment]
You said the email was sent? In that case set `SMTPDebug = 2` again, read what the output says, and read [the troubleshooting guide](https://github.com/PHPMailer/PHPMailer/wiki/Troubleshooting) to find out what to do about it.

**Comment by sadhviadigh** [comment]
<?php

use PHPMailer\PHPMailer\PHPMailer;
use PHPMailer\PHPMailer\SMTP;
use PHPMailer\PHPMailer\Exception;

require 'src/PHPMailer.php';
require 'src/SMTP.php';
require 'src/Exception.php';


/*
//require 'PHPMailerAutoload.php';
use PHPMailer\PHPMailer\PHPMailer; 
use PHPMailer\PHPMailer\Exception; 
require_once('src/PHPMailer.php'); 
require_once('src/Exception.php');*/
$mail = new PHPMailer;

$mail->isSMTP();                                      // Set mailer to use SMTP
$mail->Host = 'vsiticloud.com';  // Specify main and backup SMTP servers
$mail->SMTPAuth = true;                               // Enable SMTP authentication
$mail->Username = 'support@visticloud.com';                 // SMTP username
$mail->Password = 'Support@32';                           // SMTP password
$mail->SMTPSecure = 'tls';                            // Enable encryption, 'ssl' also accepted

$mail->From = 'support@visticloud.com';
$mail->FromName = 'Mailer';
$mail->addAddress('sadhviah@gmail.com', 'sadhvi');     // Add a recipient
$mail->addAddress('sadhviah25@gmail.com');               // Name is optional
$mail->addReplyTo('sadhviah25@gmail.com', 'Information');


$mail->WordWrap = 50;                                 // Set word wrap to 50 characters
//$mail->addAttachment('/var/tmp/file.tar.gz');         // Add attachments
//$mail->addAttachment('/tmp/image.jpg', 'new.jpg');    // Optional name
$mail->isHTML(true);                                  // Set email format to HTML

$mail->Subject = 'Here is the subject';
$mail->Body    = 'This is the HTML message body <b>in bold!</b>';
$mail->AltBody = 'This is the body in plain text for non-HTML mail clients';

if(!$mail->send()) {
    echo 'Message could not be sent.';
    echo 'Mailer Error: ' . $mail->ErrorInfo;
} else {
    echo 'Message has been sent';
}




this is my code for phpmailer, and i am getting this error:
**Message could not be sent.Mailer Error: SMTP connect() failed. https://github.com/PHPMailer/PHPMailer/wiki/Troubleshooting**

**Comment by sadhviadigh** [comment]
**Message could not be sent.Mailer Error: SMTP connect() failed. https://github.com/PHPMailer/PHPMailer/wiki/Troubleshooting**




```
<?php

use PHPMailer\PHPMailer\PHPMailer;
use PHPMailer\PHPMailer\SMTP;
use PHPMailer\PHPMailer\Exception;

require 'src/PHPMailer.php';
require 'src/SMTP.php';
require 'src/Exception.php';


/*
//require 'PHPMailerAutoload.php';
use PHPMailer\PHPMailer\PHPMailer; 
use PHPMailer\PHPMailer\Exception; 
require_once('src/PHPMailer.php'); 
require_once('src/Exception.php');*/
$mail = new PHPMailer;

$mail->isSMTP();                                      // Set mailer to use SMTP
$mail->Host = 'vsiticloud.com';  // Specify main and backup SMTP servers
$mail->SMTPAuth = true;                               // Enable SMTP authentication
$mail->Username = 'support@visticloud.com';                 // SMTP username
$mail->Password = 'Support@32';                           // SMTP password
$mail->SMTPSecure = 'tls';                            // Enable encryption, 'ssl' also accepted

$mail->From = 'support@visticloud.com';
$mail->FromName = 'Mailer';
$mail->addAddress('sadhviah@gmail.com', 'sadhvi');     // Add a recipient
$mail->addAddress('sadhviah25@gmail.com');               // Name is optional
$mail->addReplyTo('sadhviah25@gmail.com', 'Information');


$mail->WordWrap = 50;                                 // Set word wrap to 50 characters
//$mail->addAttachment('/var/tmp/file.tar.gz');         // Add attachments
//$mail->addAttachment('/tmp/image.jpg', 'new.jpg');    // Optional name
$mail->isHTML(true);                                  // Set email format to HTML

$mail->Subject = 'Here is the subject';
$mail->Body    = 'This is the HTML message body <b>in bold!</b>';
$mail->AltBody = 'This is the body in plain text for non-HTML mail clients';

if(!$mail->send()) {
    echo 'Message could not be sent.';
    echo 'Mailer Error: ' . $mail->ErrorInfo;
} else {
    echo 'Message has been sent';
}

```

**Comment by sadhviadigh** [comment]
**Message could not be sent.Mailer Error: SMTP connect() failed. https://github.com/PHPMailer/PHPMailer/wiki/Troubleshooting**
I am getting this error for the below code.

```
<?php

use PHPMailer\PHPMailer\PHPMailer;
use PHPMailer\PHPMailer\SMTP;
use PHPMailer\PHPMailer\Exception;

require 'src/PHPMailer.php';
require 'src/SMTP.php';
require 'src/Exception.php';

$mail = new PHPMailer;

$mail->isSMTP();                                      // Set mailer to use SMTP
$mail->Host = 'host@domain.com';  // Specify main and backup SMTP servers
$mail->SMTPAuth = true;                               // Enable SMTP authentication
$mail->Username = 'name@gmail.com';                 // SMTP username
$mail->Password = 'password@gmail.com';                           // SMTP password
$mail->SMTPSecure = 'tls';                            // Enable encryption, 'ssl' also accepted
$mail->Port       = 465;  

$mail->From = 'support@visticloud.com';
$mail->FromName = 'Mailer';
$mail->addAddress('sadhvi@gmail.com', 'sadhvi');     // Add a recipient
$mail->addAddress('sadhvi@gmail.com');               // Name is optional
$mail->addReplyTo('sadhvia@gmail.com', 'Information');


$mail->WordWrap = 50;                                 // Set word wrap to 50 characters
$mail->isHTML(true);                                  // Set email format to HTML

$mail->Subject = 'Here is the subject';
$mail->Body    = 'This is the HTML message body <b>in bold!</b>';
$mail->AltBody = 'This is the body in plain text for non-HTML mail clients';

if(!$mail->send()) {
    echo 'Message could not be sent.';
    echo 'Mailer Error: ' . $mail->ErrorInfo;
} else {
    echo 'Message has been sent';
}

```

**Comment by Sweepr** [comment]
Really annoying when they go bump in the night, did you rtfm?

Stop hijacking closed topics.
